mobile/src/main/java/org/openhab/habdroid/ui/OpenHABPreferencesActivity.java

```diff
@@ -139,6 +142,27 @@ public boolean onPreferenceClick(Preference preference) {
                     return true;
                 }
             });
+
+            clearCachePreference.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+                @Override
+                public boolean onPreferenceClick(Preference preference) {
+                    if (getActivity() == null) {
+                        return false;
+                    }
+                    // Get launch intent for application
+                    Intent restartIntent = getActivity().getBaseContext().getPackageManager()
+                            .getLaunchIntentForPackage(getActivity().getBaseContext().getPackageName());
```

Comment:
No need for getBaseContext() ... package manager stuff isn't theme dependent.

Reply:
Good catch (haven't read the code, just copied it from main activity) :)